source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+//	AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(0));
+//	if(asScalarDataHolder && asScalarDataHolder->dataArray().constVectorQint32().count() != ){
+//		qDebug() << "FATAL ERROR, MISMATCH BETWEEN RECEIVED DATA SIZE OF SCALER AND REQUESTED SIZE. RECEIVED: " << asScalarDataHolder->dataArray().constVectorQint32().count() << "REQUESTED: " << scalerChannelVectors.count();
+//		return;
+//	}
+
+	for(int x = 0; x < totalCount; x++){
+		AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(x));
+		if(asScalarDataHolder){
+			QVector<qint32> oneVector = asScalarDataHolder->dataArray().constVectorQint32();
+
+			int tempRunningSum;
+			QString channelString;
+			for(int y = 0, ySize = oneVector.count(); y < ySize; y++){
+				channelString = scalerChannelIndexMap.value(y);
+				tempRunningSum = scalerChannelRunningSums.value(channelString);
+				tempRunningSum += oneVector.at(y);
+				scalerChannelRunningSums[channelString] = tempRunningSum;
+
+				if( (((x+1)*baseScalerTimeScale) % largestBaseTimeScale) == 0){
+					int rebaseIndex = (x*baseScalerTimeScale)/largestBaseTimeScale;
+					(scalerChannelVectors[channelString])[rebaseIndex] = scalerChannelRunningSums.value(channelString);
+					scalerChannelRunningSums[channelString] = 0;
+				}
+			}
+		}
+	}
+
+	AMDSClientDataRequest *oneAmptekDataRequest = clientDataRequestMap_.value("Amptek SDD 240");
+
+	int initiateMovementIndex = 0;
+	bool foundMovementStart = false;
+	bool foundMovementEnd = false;
+
+	// Loop backwards from the end to find the start of the movement we're interested in
+	for(int x = oneAmptekDataRequest->data().count()-1; (x > 0) && !foundMovementStart; x--){
+		AMDSDwellSpectralDataHolder *dataHolderAsDwellSpectral = qobject_cast<AMDSDwellSpectralDataHolder*>(oneAmptekDataRequest->data().at(x));
+		int encoderPulsesInPeriod = dataHolderAsDwellSpectral->dwellStatusData().generalPurposeCounter();
+		if(!foundMovementEnd && encoderPulsesInPeriod > 20){
+			qDebug() << "Found movement end at index " << x;
+			foundMovementEnd = true;
+		}
+		else if(foundMovementEnd && encoderPulsesInPeriod < 1){
+			foundMovementStart = true;
+			initiateMovementIndex = x;
+			qDebug() << "Found movement start at index " << initiateMovementIndex;
+		}
+	}
+
+
+	int scalerInitiateMovementIndex = 0;
+	bool foundScalerMovementStart = false;
+	bool foundScalerMovementEnd = false;
+
+	QVector<qint32> encoderUpVector = scalerChannelVectors.value("EncoderUp");
+	QVector<qint32> encoderDownVector = scalerChannelVectors.value("EncoderDown");
+
+	if(double(continuousConfiguration_->scanAxes().at(0)->regionAt(0)->regionStart()) < double(continuousConfiguration_->scanAxes().at(0)->regionAt(0)->regionEnd())){
```

Comment:
Yeah, I thought this could be optimized, but I wasn't sure what the best way to do it was.